src/test-project/src/test/java/com/redhat/devtools/intellij/commonUiTestLibrary/fixturesTest/mainIdeWindow/toolWindowsPane/toolWindowsPaneOpenClose/MavenToolWindowsPaneTest.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.commonUiTestLibrary.fixturesTest.mainIdeWindow.toolWindowsPane.toolWindowsPaneOpenClose;
+
+import com.redhat.devtools.intellij.commonUiTestLibrary.fixtures.mainIdeWindow.toolWindowsPane.ProjectExplorer;
+import com.redhat.devtools.intellij.commonUiTestLibrary.fixtures.mainIdeWindow.toolWindowsPane.ToolWindowsPane;
+import com.redhat.devtools.intellij.commonUiTestLibrary.fixtures.mainIdeWindow.toolWindowsPane.buildToolPane.MavenBuildToolPane;
+import com.redhat.devtools.intellij.commonUiTestLibrary.utils.labels.ButtonLabels;
+import com.redhat.devtools.intellij.commonUiTestLibrary.utils.project.CreateCloseUtils;
+import com.redhat.devtools.intellij.commonUiTestLibrary.utils.testExtension.ScreenshotAfterTestFailExtension;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.time.Duration;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * MavenToolWindowsPane test
+ *
+ * @author zcervink@redhat.com
+ */
+@ExtendWith(ScreenshotAfterTestFailExtension.class)
+class MavenToolWindowsPaneTest extends AbstractToolWindowsPaneTest {
+    @BeforeAll
+    public static void prepareProject() {
+        CreateCloseUtils.createNewProject(remoteRobot, mavenProjectName, CreateCloseUtils.NewProjectType.MAVEN);
+        toolWindowsPane = remoteRobot.find(ToolWindowsPane.class, Duration.ofSeconds(10));
+    }
+
+    @BeforeEach
+    public void preparePanes() {
+        if (isPaneOpened(ProjectExplorer.class)) {
+            closePane(ButtonLabels.projectStripeButtonLabel, ProjectExplorer.class);
+        }
+        if (isPaneOpened(MavenBuildToolPane.class)) {
+            closePane(ButtonLabels.mavenStripeButtonLabel, MavenBuildToolPane.class);
+        }
+    }
+
+    @Test
+    public void mavenBuildToolPaneOpenCloseTest() {
+        toolWindowsPane.openMavenBuildToolPane();
+        assertTrue(isPaneOpened(MavenBuildToolPane.class), "The 'Maven Build Tool Pane' should be opened but is closed.");
+        toolWindowsPane.closeMavenBuildToolPane();
+        assertFalse(isPaneOpened(MavenBuildToolPane.class), "The 'Maven Build Tool Pane' should be closed but is opened.");
+    }
+
+    @Test
+    public void projectExplorerPaneOpenCloseTest() {
```

Comment:
Seems like project explorer pane test could be placed into different test class than in maven pane test? Maybe add toolWindowpanetest having tested project explorer open/close and missing stripeButton?

Reply:
Fixed